Number of nodes getting upgraded simultaneously also depend on surge … (#4457)
MicrosoftDocs/architecture-center: docs/operator-guides/aks/aks-upgrade-practices.md

When a node image is updated, a *cordon and drain* action is triggered on the target node pool's nodes:

> - Depending on the surge value, at a time a batch of nodes with the updated image are added to the node pool.

Reviewer: ```suggestion
- A node with the updated image is added to the node pool. The number of nodes added at the same time is governed by the surge value.
```

- Depending on the surge value, a batch of existing nodes are *cordoned* and *drained*. Cordoning ensures that the node doesn't schedule pods. Draining removes its pods and schedules them to other nodes.
- After these nodes are fully drained, they are removed from the node pool. The updated nodes added by the surge replace them.

> - This process is repeated for each node that needs to be updated in the node pool.

Reviewer: ```suggestion
- This process is repeated for each remaining batch of nodes that needs to be updated in the node pool.
```
